test/tests/functional/server_tests.js

```diff
@@ -848,19 +846,19 @@ describe('Server tests', function() {
       Connection.enableConnectionAccounting();
 
       this.configuration.manager.restart(true).then(function() {
-        locateAuthMethod(this.configuration, function(err, method) {
+        locateAuthMethod(self.configuration, function(err, method) {
           expect(err).to.be.null;
 
           // Attempt to connect
-          executeCommand(this.configuration, 'admin', {
+          executeCommand(self.configuration, 'admin', {
```

Comment:
Didn't catch this earlier because they're `auth` tests, so they didn't run at all when I tested `single`, `replicaset`, and `sharded` environments